### DISCUSSION TITLE
Clicking on some of the Eclipse toolbar icons when using dark mode in Eclipse on Windows 10

### DISCUSSION THREAD

**Discussion by marioja** [eclipse-platform/eclipse.platform.swt#765] [Q&A] [answered]
If you look at this [issue 624](https://github.com/eclipse-platform/eclipse.platform/issues/624) in the Eclipse platform you can find that in Eclipse 2023-03 the icon highlighting would not change when click on icons that toggle a behavior.  It looks like this has been fixed in Eclipse 2023-06.  Since the problem was occurring in multiple view, I think it may have been related to the SWT instead but I am not sure.

What I am looking for is the git commit that fixed this problem and some ideas or suggestions on how this could be backported to Eclipse 2023-03 even if only locally for my team since we are unable to go to Eclipse 2023-06 for the moment.

**Comment by Phillipus** [comment, marked as answer]
Might be this one https://github.com/eclipse-platform/eclipse.platform.swt/commit/fbd70cd1cce60500699938e8525a3f345bd4dc70

**Reply by marioja** [reply]
This is the commit indeed.  Are there some instructions on how I could back port the fix into 2023-03.  I was able to half bake this in my local eclipse workspace but I am wondering if I could this using the eclipse infra (github actions/jenkins ci) so that I can get an updated p2 to install locally.  I had issue with signature, hash etc.... using my half baked solution.  If it were possible to add it to the eclipse.org 2023-03 p2 that would be even greater.

**Reply by marioja** [reply]
@Phillipus Hi Phil, can you provide any insight into my question last week (Aug 18, 7:55 EDT).  Or anyone else for that matter? I had to revert to creating by hand a new org.eclipse.swt.win32.win32.x86_64_3.123.0.v20230220-1431.jar with a replaced ToolBar.class.  This cannot be put in the .p2 repository used by my installation since the hash is wrong and they are checked when installing the plugins so I have to copy this to my target .p2 used by the product setup (usually in the windows user profile).  This is an ugly hack.
